design/baremetal-operator/configmap_for_host_inspection_data.md

```diff
@@ -0,0 +1,148 @@
+reference to the owner object BareMetalHost. The data part of ConfigMap will contain
+the same hardware information as currently under `status.hardware` of BareMetalHost.
+
+### Pivoting
+
+[`clusterctl.cluster.x-k8s.io/move-hierarchy`](https://github.com/kubernetes-sigs/cluster-api/blob/0e6428151f2a388b98d1e220a698b85940a5f290/cmd/clusterctl/api/v1alpha3/labels.go#L38) label should be set on BareMetalHost CRD before issuing
+`clusterctl move` so that the entire hierarchy (i.e. including secrets, ConfigMaps)
+is pivoted to the target cluster. Having inspection data in a ConfigMap with
+ownerReferences ensures that the inspection data is available in the target cluster.
+
+### Implementation Details/Notes/Constraints
+
+The life of a ConfigMap is tightly coupled to the corresponding BareMetalHost.
+Creating or deleting a BareMetalHost will result in creation or deletion of a
+ConfigMap respectively.
+
+**Note:** [cross-namespace owner references are not allowed](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications). Thus, ConfigMap and BareMetalHost
+must be in the same namespace.
+
+- BareMetalHost is created
+- Operator creates a ConfigMap with the same name and namespace as BareMetalHost
+    and sets ownerReferences
+- Operator writes name of the ConfigMap to `spec.hardwareInfo` of the BareMetalHost
+- Provisioning requested, operator fills out the ConfigMap with the host
+    inspection data
+- Deprovisioning requested, operator cleans up the inspection data from the
+    ConfigMap
+- BareMetalHost deletion requested, operator deletes the ConfigMap and BareMetalHost
+
+#### BareMetalHost API changes
+
+- `status.hardware` field should be removed ⚠️
+- `spec.hardwareInfo` field should be added, to keep the name of ConfigMap as a
+    reference
+
+### Risks and Mitigations
+
+If an unintended user gains an access to the cluster and deletes the ConfigMap,
+BMO will notice the absence of the ConfigMap in the next reconciliation loop,
+and will try to recreate it. This will potentially trigger another provisioning.
+
+### Work Items
+
+- Remove `hardware` field from the Status of BareMetalHost API
+- Add `hardwareInfo` to the Spec of BareMetalHost API
+- Modify the operator code
+- Clean up statusAnnotation related code from BMO and CAPM3
+- Add unit tests
+- Update documentation
+- Update dependencies in Cluster-api-provider-metal3, metal3-dev-env GitHub
+    repositories
+- Give a notice to the community in advance about the change, because some
+    external code bases might have tight dependencies
+
+### Dependencies
+
+None
+
+### Test Plan
+
+Unit tests should be added for new functions and integration test scripts
+should be updated.
+
+### Upgrade / Downgrade Strategy
+
+From a user perspective, this feature doesn't introduce any breaking changes
+to existing deployments, except when there is reliance on the `status.hardware`
+field from external code base.
```

Comment:
I don't agree here. Removing the HardwareDetails field is definitely a breaking change, and not populating it any more would be a breaking change for consumers of the API.